### ISSUE TITLE
Display graph of AdSense key metrics over time in AdSense module page

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#1921]
As a follow-up to #317, the "Performance over the last 28 days" widget on the AdSense module page, which right now only contains aggregate numbers for Earnings, Page RPM, and Impressions, should be expanded to also include a graph for this information over time.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The "Performance over the last 28 days" widget on the AdSense module page should include graphs for the 3 different metrics and behave similarly like "Audience overview for the last 28 days" widget on the Analytics module page (including the #1836 fix).
    * Each of the four metrics can be clicked, showing the respective graph below.
    * Each graph should respect the currently selected date range.
    * By default, the "Earnings" graph should be active.

## Implementation Brief

<details>
<summary><strong>Context/Reference</strong></summary>

The Analytics audience overview dashboard widget is composed of 2 separate components

* The [top component](https://github.com/google/site-kit-wp/blob/21464e98e574d80721aaf4551096e9acd915bf2f/assets/js/modules/analytics/components/dashboard/AnalyticsDashboardWidgetOverview.js) shows the aggregate values and controls the active plots on the chart below it  
![image](https://user-images.githubusercontent.com/1621608/97967586-1819b480-1dc6-11eb-8844-e5f23728d13e.png)
* The [bottom component](https://github.com/google/site-kit-wp/blob/21464e98e574d80721aaf4551096e9acd915bf2f/assets/js/modules/analytics/components/dashboard/AnalyticsDashboardWidgetSiteStats.js) shows the chart with line graphs for the active plots
![image](https://user-images.githubusercontent.com/1621608/97967628-27006700-1dc6-11eb-8c4e-3e12d283e719.png)
* Both components are implemented as class-components using the legacy `withData` HOC, using a total of 3 different queries per date range
    - `overviewReportDataDefaults`
    - `userReportDataDefaults`
    - `siteAnalyticsReportDataDefaults`
* The two components are rendered as children of the main dashboard widget for the module  
https://github.com/google/site-kit-wp/blob/21464e98e574d80721aaf4551096e9acd915bf2f/assets/js/modules/analytics/components/dashboard/AnalyticsDashboardWidget.js#L198-L209

</details>

#### Overview

* Create an `AdSenseDashboardWidgetOverview` similar to `AnalyticsDashboardWidgetOverview` (close to the current `AdSensePerformanceWidget`)
* Create an `AdSenseDashboardWidgetSiteStats` similar to `AnalyticsDashboardWidgetSiteStats`
* Extract main part of `AdSensePerformanceWidget` to a new `AdSenseDashboardWidgetOverview` component
* Update `AdSensePerformanceWidget` to wrap `AdSenseDashboardWidgetOverview` and `AdSenseDashboardWidgetSiteStats` components
* Use functional components for all pieces
* Use datastore for all requests, error & zero-data handling/awareness (**no `withData`**)
    - Better to not create new components using legacy batch architecture where possible
    - The AdSense dashboard only makes a few requests so it's less critical to leverage batching here (yet)
* The names of the new components could change, it's defined like this for now for consistency

#### `AdSenseDashboardWidgetOverview`

* Create `assets/js/modules/adsense/components/dashboard/AdSenseDashboardWidgetOverview.js`
* Re-implement queries using `getReport` selector using the relevant options
    - select `getDateRangeDates( { compare: true } )` from `core/user` to get the dates to use for both last and previous date ranges
        - returned `{ startDate, endDate, compareStartDate, compareEndDate }` are referenced below
    - `const metrics = [ 'EARNINGS', 'PAGE_VIEWS_RPM', 'IMPRESSIONS', 'PAGE_VIEWS_CTR' ];`
    - select `getReport( { metrics, startDate, endDate } )` as `currentRangeData`
    - select `getReport( { metrics, startDate: compareStartDate, endDate: compareEndDate } )` as `prevRangeData`
    - Use alternate `isDataZeroAdSense` (or update if no other uses) to no longer be coupled specifically to `last-28-days`
        - Use the new utility (yet to be introduced) from #2242 **if available** (otherwise `TODO` the workaround if needed)
        - Zero data state should only be based on the current range data, not previous
    - Select `isLoading` based on whether `getReport` is currently resolving for either selector
        - if loading, return `<PreviewBlock width="100%" height="250px" />`
    - Select errors for the reports using `getErrorForSelector( 'getReport', [options] )` from the Adsense store
        - Return error component if present using `getDataErrorComponent` (more on this below)
* Accept `selectedStats` and `handleStatSelection` (function) as props to be passed through to the inner `DataBlock` components
    - These are optional (the current performance widget passes these through to the `DataBlock` but they're undefined) but necessary for the composite behavior
* Update `dataBlocks` definitions to include a `selected` property, based on presence in `selectedStats`
    - Default to the Earnings block selected if none provided

#### `AdSenseDashboardWidgetSiteStats`

* Scaffold out based on [`AnalyticsDashboardWidgetSiteStats`](https://github.com/google/site-kit-wp/blob/21464e98e574d80721aaf4551096e9acd915bf2f/assets/js/modules/analytics/components/dashboard/AnalyticsDashboardWidgetSiteStats.js) as functional component (basically a `render` function as is)
* No reason for `setOptions` to be its own function, it's mostly static as-is
    - no `pageTitle`
    - potentially extract to a common definition in utils
* Require `selectedStats` prop to be passed through to the inner `GoogleChart` component
* select data with `getReport` selector
* Select `isLoading` based on whether `getReport` is currently resolving for either selector
    - if loading, return `<PreviewBlock width="100%" height="250px" />`
* Select errors for the reports using `getErrorForSelector( 'getReport', [options] )` from the Adsense store
    - Return error component if present using `getDataErrorComponent`
* [Define `series` and `vAxes`](https://github.com/google/site-kit-wp/blob/21464e98e574d80721aaf4551096e9acd915bf2f/assets/js/modules/analytics/components/dashboard/AnalyticsDashboardWidget.js#L118-L141) internally
    - No reason to accept as props as in `AnalyticsDashboardWidget` since the value doesn't need to be variable

#### `AdSensePerformanceWidget`

* Replace current implementation with new `export default function AdSensePerformanceWidget() {}`
* [`useState` to manage selected stats](https://github.com/google/site-kit-wp/blob/21464e98e574d80721aaf4551096e9acd915bf2f/assets/js/modules/analytics/components/dashboard/AnalyticsDashboardWidget.js#L56)
    - Default to Earnings as initially selected
* Render inner components, passing through props
  ```js
  <AdSenseDashboardWidgetOverview 
    selectedStats={ selectedStats } // from useState
    handleStatSelection={ handleStatSelection } // Function to call setSelectedStats
  />
  <AdSenseDashboardWidgetSiteStats 
    selectedStats={ selectedStats } // from useState
  />
  ```
* Port handling of props passed in from `AdSenseDashboardWidget` for communicating back to the parent
  - Add a `useEffect` call after select hooks which depends on their data
    - Annotate this with a `TODO` comment that this is to be removed when `AdSenseDashboardWidget` is refactored
  - If the reports have finished loading, and there is no error, call `props.handleDataSuccess` with no args
  - If the reports have finished loading, and there **is** an error, call `props.handleDataError` with the `error.message` and `error` positional arguments
  - If the reports have finished loading, and there is no data, call `props.handleDataError` with with no args
  - Only data from `AdSenseDashboardWidgetOverview` should be considered for the above handler functions

#### Error & No Data States
* Each inner component (`AdSenseDashboardWidgetOverview` and `AdSenseDashboardWidgetSiteStats`) would independently render an error and no-data state
* `AdSensePerformanceWidget` would also select for errors / no data using logic defined with the child components
  - If any error was returned, a single error/no-data component would be rendered (rather than one each)
  - To do this without coupling the parent component to child component selector options, each child component needs to expose a function which can be provided to `useSelect` which would select whatever it needs and returns a boolean `hasData` (or undefined if unresolved)  
  This can be provided right on the component, just like `defaultProps` and `propTypes` are "attached" to a component
      - E.g. `ComponentFunction.selectHasData`
  - Similarly, each child component would expose a function which can be provided to `useSelect` that selects whatever it needs and returns a `reportError` or `null` if none
      - E.g. `ComponentFunction.selectReportError`
  - Expose a function which can be provided to `useSelect` that returns a boolean `isLoading` if any of its reports are not yet resolved
      - E.g. `ComponentFunction.selectIsLoading`
  - This ensures that error and no-data states are scoped to the current date range
  - e.g.  
  ```js
  function AdSensePerformanceWidget() {
    const overviewHasData = useSelect( AdSenseDashboardWidgetOverview.selectHasData );
    const overviewReportError = useSelect( AdSenseDashboardWidgetOverview.selectReportError );
    const statsHasData = useSelect( AdSenseDashboardWidgetSiteStats.selectHasData );
    const statsReportError = useSelect( AdSenseDashboardWidgetSiteStats.selectReportError );
    const isZeroData = ! ( overviewHasData && statsHasData );
    const error = overviewReportError || statsReportError;

    if ( error ) {
      return // Error component
    } else if ( isZeroData ) {
      return // Zero data component
    }
    return (
      <Fragment>
        { /* Just add props! */ }
        <AdSenseDashboardWidgetOverview />
        <AdSenseDashboardWidgetSiteStats />
      </Fragment>
    )
  }
  ```

#### Charting the data

<details>
<summary>Analytics uses a format like this to pass its data to the <code>GoogleChart</code></summary>

e.g. using `last-7-days`
```js
[
  [
    {
      "type": "date",
      "label": "Day"
    },
    {
      "type": "string",
      "role": "tooltip",
      "p": {
        "html": true,
        "role": "tooltip"
      }
    },
    {
      "type": "number",
      "label": "Users"
    },
    {
      "type": "number",
      "label": "Previous period"
    }
  ],
  [
    "2020-10-26T22:00:00.000Z",
    "<div class=\"googlesitekit-visualization-tooltip googlesitekit-visualization-tooltip--down\">\n\t\t\t\t<p>Tue, Oct 27 vs Tue, Oct 20</p>\n\t\t\t\t<p>Users: <strong>43</strong> <em><svg width=\"9\" height=\"9\" viewBox=\"0 0 10 10\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\" class=\"googlesitekit-change-arrow googlesitekit-change-arrow--down\">\n\t\t\t\t<path d=\"M5.625 10L5.625 2.375L9.125 5.875L10 5L5 -1.76555e-07L-2.7055e-07 5L0.875 5.875L4.375 2.375L4.375 10L5.625 10Z\" fill=\"currentColor\" />\n\t\t\t</svg> 2.27%</em></p>\n\t\t\t</div>",
    "43",
    "44"
  ],
  [
    "2020-10-27T22:00:00.000Z",
    "<div class=\"googlesitekit-visualization-tooltip googlesitekit-visualization-tooltip--up\">\n\t\t\t\t<p>Wed, Oct 28 vs Wed, Oct 21</p>\n\t\t\t\t<p>Users: <strong>43</strong> <em><svg width=\"9\" height=\"9\" viewBox=\"0 0 10 10\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\" class=\"googlesitekit-change-arrow googlesitekit-change-arrow--up\">\n\t\t\t\t<path d=\"M5.625 10L5.625 2.375L9.125 5.875L10 5L5 -1.76555e-07L-2.7055e-07 5L0.875 5.875L4.375 2.375L4.375 10L5.625 10Z\" fill=\"currentColor\" />\n\t\t\t</svg> 53.57%</em></p>\n\t\t\t</div>",
    "43",
    "28"
  ],
  [
    "2020-10-28T22:00:00.000Z",
    "<div class=\"googlesitekit-visualization-tooltip googlesitekit-visualization-tooltip--up\">\n\t\t\t\t<p>Thu, Oct 29 vs Thu, Oct 22</p>\n\t\t\t\t<p>Users: <strong>47</strong> <em><svg width=\"9\" height=\"9\" viewBox=\"0 0 10 10\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\" class=\"googlesitekit-change-arrow googlesitekit-change-arrow--up\">\n\t\t\t\t<path d=\"M5.625 10L5.625 2.375L9.125 5.875L10 5L5 -1.76555e-07L-2.7055e-07 5L0.875 5.875L4.375 2.375L4.375 10L5.625 10Z\" fill=\"currentColor\" />\n\t\t\t</svg> 17.5%</em></p>\n\t\t\t</div>",
    "47",
    "40"
  ],
  [
    "2020-10-29T22:00:00.000Z",
    "<div class=\"googlesitekit-visualization-tooltip googlesitekit-visualization-tooltip--up\">\n\t\t\t\t<p>Fri, Oct 30 vs Fri, Oct 23</p>\n\t\t\t\t<p>Users: <strong>33</strong> <em><svg width=\"9\" height=\"9\" viewBox=\"0 0 10 10\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\" class=\"googlesitekit-change-arrow googlesitekit-change-arrow--up\">\n\t\t\t\t<path d=\"M5.625 10L5.625 2.375L9.125 5.875L10 5L5 -1.76555e-07L-2.7055e-07 5L0.875 5.875L4.375 2.375L4.375 10L5.625 10Z\" fill=\"currentColor\" />\n\t\t\t</svg> 3.13%</em></p>\n\t\t\t</div>",
    "33",
    "32"
  ],
  [
    "2020-10-30T22:00:00.000Z",
    "<div class=\"googlesitekit-visualization-tooltip googlesitekit-visualization-tooltip--down\">\n\t\t\t\t<p>Sat, Oct 31 vs Sat, Oct 24</p>\n\t\t\t\t<p>Users: <strong>10</strong> <em><svg width=\"9\" height=\"9\" viewBox=\"0 0 10 10\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\" class=\"googlesitekit-change-arrow googlesitekit-change-arrow--down\">\n\t\t\t\t<path d=\"M5.625 10L5.625 2.375L9.125 5.875L10 5L5 -1.76555e-07L-2.7055e-07 5L0.875 5.875L4.375 2.375L4.375 10L5.625 10Z\" fill=\"currentColor\" />\n\t\t\t</svg> 61.54%</em></p>\n\t\t\t</div>",
    "10",
    "26"
  ],
  [
    "2020-10-31T22:00:00.000Z",
    "<div class=\"googlesitekit-visualization-tooltip\">\n\t\t\t\t<p>Sun, Nov 1 vs Sun, Oct 25</p>\n\t\t\t\t<p>Users: <strong>20</strong> <em><svg width=\"9\" height=\"9\" viewBox=\"0 0 10 10\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\" class=\"googlesitekit-change-arrow\">\n\t\t\t\t<path d=\"M5.625 10L5.625 2.375L9.125 5.875L10 5L5 -1.76555e-07L-2.7055e-07 5L0.875 5.875L4.375 2.375L4.375 10L5.625 10Z\" fill=\"currentColor\" />\n\t\t\t</svg> 0%</em></p>\n\t\t\t</div>",
    "20",
    "20"
  ],
  [
    "2020-11-01T22:00:00.000Z",
    "<div class=\"googlesitekit-visualization-tooltip googlesitekit-visualization-tooltip--up\">\n\t\t\t\t<p>Mon, Nov 2 vs Mon, Oct 26</p>\n\t\t\t\t<p>Users: <strong>46</strong> <em><svg width=\"9\" height=\"9\" viewBox=\"0 0 10 10\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\" class=\"googlesitekit-change-arrow googlesitekit-change-arrow--up\">\n\t\t\t\t<path d=\"M5.625 10L5.625 2.375L9.125 5.875L10 5L5 -1.76555e-07L-2.7055e-07 5L0.875 5.875L4.375 2.375L4.375 10L5.625 10Z\" fill=\"currentColor\" />\n\t\t\t</svg> 35.29%</em></p>\n\t\t\t</div>",
    "46",
    "34"
  ]
]
```
</details>

* See [`extractAnalyticsDashboardData`](https://github.com/google/site-kit-wp/blob/21464e98e574d80721aaf4551096e9acd915bf2f/assets/js/modules/analytics/util/index.js#L112) to see how this is prepared
    - Use many small composable utilities rather than one big utility to avoid coupling to the response format
    - A single composite function can be defined/used in the component but should be easy to change (e.g. if we change the metrics returned)

### Test Coverage

* Add Storybook stories for each component
    - `AdSenseDashboardWidgetOverview`
    - `AdSenseDashboardWidgetSiteStats`
* Update story for the current `AdSensePerformanceWidget`
* Could add a few Jest tests if time allows but not critical

### Visual Regression Changes

* None expected

## QA Brief

* Set up the plugin on your site;
* Activate and connect the AdSense module using an account that has real non-zero data;
* Go to the AdSense page and check that you see the graph with AdSense statistic information;
* That graph should function similarly to the graph we have on the Analytics page;
  * Make sure you can switch between different metrics (earnings, page rpm, impressions, page ctr) and the graph re-renders data accordingly;
  * Make sure that when you hover over points on the graph, you see appropriate tooltips.
* Connect another AdSense account that has no live data and make sure that you see the appropriate message instead of the graph.

## Changelog entry

* Display graph with historic AdSense data on module page, showing earnings, RPM, impressions and CTR over time.

**Comment by felixarntz** [comment]
@aaemnnosttv The IB on this reads really solid! Just a few thoughts:

One thing that I think (unfortunately) needs to be maintained and is currently missing from here is continued support of the `handleDataSuccess` and `handleDataError` props which `AdSenseDashboardWidget` passes to `AdSensePerformanceWidget`. These props are crucial for conditionally adjusting the entire AdSense screen in case there's no data available. The approach is quite confusing, but we shouldn't get into refactoring it `AdSenseDashboardWidget` here IMO.

These props are currently handled via the `withData` HOC that `AdSensePerformanceWidget` uses. But since we now switch to using the datastore, we'll need to maintain this a bit more manually, probably via `useEffect` hooks. Essentially as follows:
* If error, call `handleDataError` with the error message as first parameter and the full error as second parameter. 🤦‍♂️ 
* If zero data, call `handleDataError` without any parameter. 🤦‍♂️ 
* If no error and no zero data, call `handleDataSuccess` without any parameter.

These prop functions should solely be based on the _current_ date range data in the `AdSenseDashboardWidgetOverview` component you're describing - previous data to compare with, as well as the date-specific data from the other component shouldn't affect this.

We don't need to implement this in a nice way. Let's just pass through the two props to where they need to be used and add `TODO` comments to remove these as part of refactoring `AdSenseDashboardWidget`.

> Use alternate `isDataZeroAdSense` (or update if no other uses) to no longer be coupled specifically to `last-28-days`

See related #2242 which will introduce a better version of this. I think the work here can happen in parallel, but we should preferably use the function from that issue once it's ready (which due to the size of this will likely happen sooner anyway).

**Comment by aaemnnosttv** [comment]
> One thing that I think (unfortunately) needs to be maintained and is currently missing from here is continued support of the `handleDataSuccess` and `handleDataError` props which `AdSenseDashboardWidget` passes to `AdSensePerformanceWidget`. These props are crucial for conditionally adjusting the entire AdSense screen in case there's no data available.

Ah yes, good catch; I'll add this to the IB 👍 

> These prop functions should solely be based on the _current_ date range data in the `AdSenseDashboardWidgetOverview` component you're describing - previous data to compare with, **as well as the date-specific data from the other component** shouldn't affect this.

Do you mean that an error or zero-data in `AdSenseDashboardWidgetSiteStats` would not trigger the given handler? I've updated the IB assuming this is what you meant but just wanted to make sure.

> See related #2242 which will introduce a better version of this.

Updated this part to use this instead if available or mark it with a TODO if needed.

---

IB updated 👍

**Comment by felixarntz** [comment]
IB ✅  @aaemnnosttv 
> Do you mean that an error or zero-data in `AdSenseDashboardWidgetSiteStats` would not trigger the given handler? I've updated the IB assuming this is what you meant but just wanted to make sure.

Yes, that's correct.

**Comment by felixarntz** [comment]
@eugene-manuilov Can you still add a QA Brief to this one?

**Comment by eugene-manuilov** [comment]
QA Brief is added. Assigning it to @wpdarren.

**Comment by wpdarren** [comment]
### QA update:  Pass ✅ 

@eugene-manuilov I have passed this but an observation: when you select 90 days the earning and Page RPM values get difficult to read due to the number of characters and font size. I suspect that this is outside of the scope for this ticket, but do you think we should create a new ticket for it?  - [Screenshot](https://user-images.githubusercontent.com/73545194/101642208-af30f680-3a6d-11eb-8f75-103b89d08075.png)

**Verified:**

- The AdSense page displays the graph with AdSense statistic information  - [Screenshot](https://user-images.githubusercontent.com/73545194/101640715-cd95f280-3a6b-11eb-916d-18ed080732e2.png)
- Selected 7, 14, 28 and 90 day in the date picker and checked AdSense statistic information - [Screenshot](https://user-images.githubusercontent.com/73545194/101640735-d4246a00-3a6b-11eb-8ca7-e74c69c0c073.png)
- When you switch between different metrics the graph re-renders data accordingly - [Screenshot](https://user-images.githubusercontent.com/73545194/101642608-26668a80-3a6e-11eb-934a-37cf410b2d7a.png)
- When you hover over points on the graph the appropriate tooltips are displayed - [Screenshots](https://user-images.githubusercontent.com/73545194/101642634-2e262f00-3a6e-11eb-94a5-d8a4c82a823c.png)
- On an Adsense account with no data, the message appears as expected - [Screenshot](https://user-images.githubusercontent.com/73545194/101644546-48610c80-3a70-11eb-8ddc-a003d76c2abf.png)

**Comment by eugene-manuilov** [comment]
@wpdarren yes, please, create a ticket for font styles. Also looks like data for the previous period hasn't been loaded for your charts, which is weird :thinking: If everything works correctly you should see something like this: https://google.github.io/site-kit-wp/storybook/develop/?path=/story/adsense-module--performance

Do you see one line for all date ranges or just for 90 days only?

**Comment by wpdarren** [comment]
@eugene-manuilov this is a good observation, I should have picked up on that.  The previous period data isn't appearing on any date range.  Not sure if this is why, but I am using the tester plugin because I do not have any Google Adsense accounts that have real data - [Screenshot](https://user-images.githubusercontent.com/73545194/101762507-7c900800-3b18-11eb-96b4-06bb18c61eeb.png). FYI - I have just created a new test site on develop branch and the issue still exists. 

I'm going to fail this and send it back for review.  ❌

**Comment by eugene-manuilov** [comment]
> @eugene-manuilov this is a good observation, I should have picked up on that. The previous period data isn't appearing on any date range. Not sure if this is why, but I am using the tester plugin because I do not have any Google Adsense accounts that have real data - [Screenshot](https://user-images.githubusercontent.com/73545194/101762507-7c900800-3b18-11eb-96b4-06bb18c61eeb.png). FYI - I have just created a new test site on develop branch and the issue still exists.
> 
> I'm going to fail this and send it back for review.

Ok, it makes sense then. The tester plugin doesn't provide appropriate testing data for these graphs.

@cole10up can you take a look at it?

**Comment by cole10up** [comment]
@eugene-manuilov 

The only site I have access to that has some data only has impressions and the revenue has been stagnant for months.

![image](https://user-images.githubusercontent.com/57115907/101866769-87559780-3b2e-11eb-9805-b8d54e41d6d5.png)

We'll have to ask around.

**Comment by eugene-manuilov** [comment]
@cole10up could you, please, click on the `Impressions` tab and make a screenshot of its graph?

**Comment by cole10up** [comment]
Absolutely. @eugene-manuilov 

### QA ❌

Checking impressions:

90 days is looking off for me on my site. The Y-Axis is showing a max of 6, when we're showing 20 when clicked.

![image](https://user-images.githubusercontent.com/57115907/101872577-57ac8c80-3b3a-11eb-8e40-74d53fdfa0db.png)

I think this is due to the default 28 days being 6... Therefore changing, the Y-Axis remains at 6?

![image](https://user-images.githubusercontent.com/57115907/101872669-7f035980-3b3a-11eb-999a-f558a7a54954.png)

14 days shows 6 impressions as well as 7 days
![image](https://user-images.githubusercontent.com/57115907/101872796-afe38e80-3b3a-11eb-80fe-70a5ca02b7e7.png)

![image](https://user-images.githubusercontent.com/57115907/101872812-b8d46000-3b3a-11eb-8f98-d77156b5a89a.png)

**Comment by eugene-manuilov** [comment]
@cole10up ok, based on your screenshots I think it works correctly :smile: The number in the Impressions tab is total. If you summarize all values in the graph, you will see that it equals the number in tab. 6 - the maximum of daily impressions.

**Comment by wpdarren** [comment]
@cole10up I have messaged James to find out if he has a test site with Live Adsense data. Will pick this up when we have a site to test on.  This ticket is on hold.

**Comment by wpdarren** [comment]
@eugene-manuilov @cole10up  I have looked at a site with live Adsense data and the previous period is now appearing on the graphs. I went through all of the date periods, and the appropriate data, i.e. Impressions, Rate CTR, etc. The tooltips were appearing against current and previous data.  It's looking good.  

Here are some screenshots:

[Earnings over 28 days](https://user-images.githubusercontent.com/73545194/101916185-90fbff80-3c01-11eb-9062-6d924abab309.png)

[Impressions over 90 days](https://user-images.githubusercontent.com/73545194/101916248-a2dda280-3c01-11eb-9fef-0533d12239f7.png)

[Page RPM over 14 days ](https://user-images.githubusercontent.com/73545194/101916269-aa04b080-3c01-11eb-80aa-87893c2911da.png)
